### Youth smoking drug data

The dataset provides information about smoking and drug experiments trends from 2020-2024,with approximately 10,000 records.



### The dataset features.
1.**Year**: The year of data collection.

2.**Age_Group**: Age range (e.g., 10-14, 15-19, etc.).

3.**Gender**: Gender (e.g., Male, Female, Both).

4.**Smoking_Prevalence**: Percentage of youth who smoke.

5.**Drug_Experimentation**: Percentage of youth experimenting with drugs.

6.**Socioeconomic_Status**: Categories like High, Middle.

7.**Peer_Influence**: A numeric measure of peer pressure.

8.**School_Programs**: Whether schools have anti-smoking/drug programs.

9.**Family_Background**: A numeric measure of family influence.

10.**Mental_Health**: A numeric rating of mental health.

11.**Access_to_Counseling**: Whether counseling is available.

12.**Parental_Supervision**: A numeric measure of parental involvement.

13.**Substance_Education**: Whether they received substance education.

14.**Community_Support**: A numeric rating of community support.

15.**Media_Influence**: A numeric measure of media influence.

### Key objectives:

The dataset aims to help researches,public health organizations,and policymakers understand the patterns and factors influencing substance use among youth.By identifying trends and influential factors,stakeholders can make data-driven decisions to combat substance abuse effectively.

In [31]:
import pandas as pd 

df= pd.read_csv('dataset/youth_smoking_drug_data_10000_rows_expanded.csv')

df.head()

Year Age_Group  Gender  Smoking_Prevalence  Drug_Experimentation  \
0  2024     15-19    Both               18.85                 32.40   
1  2024     10-14  Female               34.88                 41.57   
2  2023     10-14    Both               42.00                 56.80   
3  2024     40-49    Both               33.75                 42.90   
4  2023     15-19    Male               47.90                 39.62   

  Socioeconomic_Status  Peer_Influence School_Programs  Family_Background  \
0                 High               5             Yes                  1   
1                 High               6             Yes                 10   
2                 High               6             Yes                  2   
3               Middle              10              No                  9   
4                 High               1              No                  2   

   Mental_Health Access_to_Counseling  Parental_Supervision  \
0              5                   No                     4   
1              5                   No                     9   
2              7                  Yes                     2   
3              7                  Yes                     2   
4              4                  Yes                     4   

  Substance_Education  Community_Support  Media_Influence  
0                  No                  3                1  
1                 Yes                  9                3  
2                  No                  5                1  
3                  No                 10                9  
4                  No                 10                3

In [32]:
df.columns

Index(['Year', 'Age_Group', 'Gender', 'Smoking_Prevalence',
       'Drug_Experimentation', 'Socioeconomic_Status', 'Peer_Influence',
       'School_Programs', 'Family_Background', 'Mental_Health',
       'Access_to_Counseling', 'Parental_Supervision', 'Substance_Education',
       'Community_Support', 'Media_Influence'],
      dtype='object')

In [33]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10000 entries, 0 to 9999
Data columns (total 15 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   Year                  10000 non-null  int64  
 1   Age_Group             10000 non-null  object 
 2   Gender                10000 non-null  object 
 3   Smoking_Prevalence    10000 non-null  float64
 4   Drug_Experimentation  10000 non-null  float64
 5   Socioeconomic_Status  10000 non-null  object 
 6   Peer_Influence        10000 non-null  int64  
 7   School_Programs       10000 non-null  object 
 8   Family_Background     10000 non-null  int64  
 9   Mental_Health         10000 non-null  int64  
 10  Access_to_Counseling  10000 non-null  object 
 11  Parental_Supervision  10000 non-null  int64  
 12  Substance_Education   10000 non-null  object 
 13  Community_Support     10000 non-null  int64  
 14  Media_Influence       10000 non-null  int64  
dtypes: float64(2), int64

In [34]:
df.isnull().sum()

Year                    0
Age_Group               0
Gender                  0
Smoking_Prevalence      0
Drug_Experimentation    0
Socioeconomic_Status    0
Peer_Influence          0
School_Programs         0
Family_Background       0
Mental_Health           0
Access_to_Counseling    0
Parental_Supervision    0
Substance_Education     0
Community_Support       0
Media_Influence         0
dtype: int64

The dataset does not contain the null-values has 10,000 records

In [35]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10000 entries, 0 to 9999
Data columns (total 15 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   Year                  10000 non-null  int64  
 1   Age_Group             10000 non-null  object 
 2   Gender                10000 non-null  object 
 3   Smoking_Prevalence    10000 non-null  float64
 4   Drug_Experimentation  10000 non-null  float64
 5   Socioeconomic_Status  10000 non-null  object 
 6   Peer_Influence        10000 non-null  int64  
 7   School_Programs       10000 non-null  object 
 8   Family_Background     10000 non-null  int64  
 9   Mental_Health         10000 non-null  int64  
 10  Access_to_Counseling  10000 non-null  object 
 11  Parental_Supervision  10000 non-null  int64  
 12  Substance_Education   10000 non-null  object 
 13  Community_Support     10000 non-null  int64  
 14  Media_Influence       10000 non-null  int64  
dtypes: float64(2), int64

***Categorical columns***

In [36]:
categorical = ['Age_Group', 'Gender', 'Socioeconomic_Status', 'School_Programs', 'Access_to_Counseling', 'Substance_Education']

In [37]:
categorical = [var for var in df.columns if df[var].dtype=='O'] 
print('There are {} categorical variables\n'.format(len(categorical))) 
print('The categorical variables are:',categorical) 

There are 6 categorical variables

The categorical variables are: ['Age_Group', 'Gender', 'Socioeconomic_Status', 'School_Programs', 'Access_to_Counseling', 'Substance_Education']


***Converted the categorical columns to numerical values using one hot encoding to make effective model without any interruptions***

In [38]:
df[categorical].head()

Age_Group  Gender Socioeconomic_Status School_Programs Access_to_Counseling  \
0     15-19    Both                 High             Yes                   No   
1     10-14  Female                 High             Yes                   No   
2     10-14    Both                 High             Yes                  Yes   
3     40-49    Both               Middle              No                  Yes   
4     15-19    Male                 High              No                  Yes   

  Substance_Education  
0                  No  
1                 Yes  
2                  No  
3                  No  
4                  No

***checking for missing values in categorical variables***

In [39]:
df[categorical].isnull().sum()

Age_Group               0
Gender                  0
Socioeconomic_Status    0
School_Programs         0
Access_to_Counseling    0
Substance_Education     0
dtype: int64

***viewing frequency of categorical variables***

In [40]:
for var in categorical: 
    print(df[var].value_counts())

Age_Group
20-24    1045
30-39    1021
60-69    1020
15-19    1014
10-14    1007
40-49    1001
50-59    1001
80+       979
25-29     974
70-79     938
Name: count, dtype: int64
Gender
Both      3415
Male      3294
Female    3291
Name: count, dtype: int64
Socioeconomic_Status
High      3364
Middle    3344
Low       3292
Name: count, dtype: int64
School_Programs
No     5041
Yes    4959
Name: count, dtype: int64
Access_to_Counseling
No     5085
Yes    4915
Name: count, dtype: int64
Substance_Education
Yes    5011
No     4989
Name: count, dtype: int64


**Age_Group**: Slight variation between groups, but no severe imbalance.

**Gender**: Well-balanced distribution with slightly more individuals identifying as Both.

**Socioeconomic_Status**: Even distribution across High, Middle, and Low categories.

**School_Programs**: Nearly equal distribution between those with and without access. 
**Substance_Education**: Perfectly balanced.

In [41]:
#checking for cardinality in categorical values.
for var in categorical: 
    print(var, 'contains' ,len(df[var].unique()), 'labels')

Age_Group contains 10 labels
Gender contains 3 labels
Socioeconomic_Status contains 3 labels
School_Programs contains 2 labels
Access_to_Counseling contains 2 labels
Substance_Education contains 2 labels


 Age group has a moderate cardinality with 10 labels which may not be useful for the model , could raise issues when performing one hot encoding.

 Gender and Socioeconomic_Status contains 3 labels showing low cardinality that when one hot encoding is performed it wont be much affected.
 
 School_Programs,Access_to_Counselling,Substance_Education contains 2 labels indicating low cardinality this would be suitable when performing one hot encoding

### Feature Engineering

### Explore Age_group Variable

In [42]:
print('Age group contains' , len(df['Age_Group'].unique()),'label')

Age group contains 10 label


In [43]:
df['Age_Group'].unique()

array(['15-19', '10-14', '40-49', '70-79', '30-39', '60-69', '50-59',
       '25-29', '80+', '20-24'], dtype=object)

In [44]:
# checking for frequency distribution of values in Age_Group variable 
df['Age_Group'].value_counts()

Age_Group
20-24    1045
30-39    1021
60-69    1020
15-19    1014
10-14    1007
40-49    1001
50-59    1001
80+       979
25-29     974
70-79     938
Name: count, dtype: int64

The counts for each age group are relatively close, ranging from 938 to 1045. This indicates that your dataset has a fairly balanced representation of different age groups, with no age group dominating the dataset. The largest group (20-24) has 1,045 entries, and the smallest group (70-79) has 938 entries, a difference of about 107, which is not significant considering the total size.

Exploring Gender variable


In [45]:
print('Gender',len(df['Gender'].unique()),'label')

Gender 3 label


In [46]:
df['Gender'].value_counts()

Gender
Both      3415
Male      3294
Female    3291
Name: count, dtype: int64

The Male and Female groups have nearly identical counts, with 3,294 males and 3,291 females, indicating a balanced gender distribution in the dataset.

In [47]:
pd.get_dummies(df['Gender'], drop_first=False, dtype='int').head()

Both  Female  Male
0     1       0     0
1     0       1     0
2     1       0     0
3     1       0     0
4     0       0     1

Exploring Socioeconomic_Status Variable

In [48]:
print('Socioeconomic_Status',len(df['Socioeconomic_Status'].unique()),'label')

Socioeconomic_Status 3 label


In [49]:
df['Socioeconomic_Status'].value_counts()

Socioeconomic_Status
High      3364
Middle    3344
Low       3292
Name: count, dtype: int64

Balanced Distribution: The three categories—High (3,364), Middle (3,344), and Low (3,292)—are very close in count, suggesting a balanced representation of different socioeconomic statuses. The differences between them are minimal, with a gap of just 72 entries between the largest (High) and the smallest (Low) categories.

Equitable Coverage: Since all socioeconomic groups are similarly represented, the data should provide a fairly unbiased look at how different socioeconomic statuses impact outcomes such as smoking prevalence or drug experimentation.

Exploring School_Programs

In [50]:
print('School_Programs',len (df['School_Programs'].unique()),'label')

School_Programs 2 label


In [51]:
df['School_Programs'].value_counts()

School_Programs
No     5041
Yes    4959
Name: count, dtype: int64

In [52]:
print('Access_to_Counseling', len(df['Access_to_Counseling'].unique()),'label')

Access_to_Counseling 2 label


In [53]:
df['Access_to_Counseling'].value_counts()

Access_to_Counseling
No     5085
Yes    4915
Name: count, dtype: int64

In [54]:
print('Substance_Education',len(df['Substance_Education'].unique()),'label')

Substance_Education 2 label


In [55]:
df['Substance_Education'].value_counts()

Substance_Education
Yes    5011
No     4989
Name: count, dtype: int64

In [56]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10000 entries, 0 to 9999
Data columns (total 15 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   Year                  10000 non-null  int64  
 1   Age_Group             10000 non-null  object 
 2   Gender                10000 non-null  object 
 3   Smoking_Prevalence    10000 non-null  float64
 4   Drug_Experimentation  10000 non-null  float64
 5   Socioeconomic_Status  10000 non-null  object 
 6   Peer_Influence        10000 non-null  int64  
 7   School_Programs       10000 non-null  object 
 8   Family_Background     10000 non-null  int64  
 9   Mental_Health         10000 non-null  int64  
 10  Access_to_Counseling  10000 non-null  object 
 11  Parental_Supervision  10000 non-null  int64  
 12  Substance_Education   10000 non-null  object 
 13  Community_Support     10000 non-null  int64  
 14  Media_Influence       10000 non-null  int64  
dtypes: float64(2), int64

In [58]:
df_encoded = pd.get_dummies(df, columns=categorical, drop_first=False)
df_encoded.head()

Year  Smoking_Prevalence  Drug_Experimentation  Peer_Influence  \
0  2024               18.85                 32.40               5   
1  2024               34.88                 41.57               6   
2  2023               42.00                 56.80               6   
3  2024               33.75                 42.90              10   
4  2023               47.90                 39.62               1   

   Family_Background  Mental_Health  Parental_Supervision  Community_Support  \
0                  1              5                     4                  3   
1                 10              5                     9                  9   
2                  2              7                     2                  5   
3                  9              7                     2                 10   
4                  2              4                     4                 10   

   Media_Influence  Age_Group_10-14  ...  Gender_Male  \
0                1            False  ...        False   
1                3             True  ...        False   
2                1             True  ...        False   
3                9            False  ...        False   
4                3            False  ...         True   

   Socioeconomic_Status_High  Socioeconomic_Status_Low  \
0                       True                     False   
1                       True                     False   
2                       True                     False   
3                      False                     False   
4                       True                     False   

   Socioeconomic_Status_Middle  School_Programs_No  School_Programs_Yes  \
0                        False               False                 True   
1                        False               False                 True   
2                        False               False                 True   
3                         True                True                False   
4                        False                True                False   

   Access_to_Counseling_No  Access_to_Counseling_Yes  Substance_Education_No  \
0                     True                     False                    True   
1                     True                     False                   False   
2                    False                      True                    True   
3                    False                      True                    True   
4                    False                      True                    True   

   Substance_Education_Yes  
0                    False  
1                     True  
2                    False  
3                    False  
4                    False  

[5 rows x 31 columns]

### Frequency of categorical features

In [ ]:
import seaborn as sns 
